activerecord/lib/active_record/attribute_methods/dirty.rb

```diff
@@ -229,7 +228,13 @@ def attribute_names_for_partial_updates
         end
 
         def attribute_names_for_partial_inserts
-          partial_inserts? ? changed_attribute_names_to_save : attribute_names
+          partial_inserts? ? changed_attribute_names_to_save : attribute_names_without_changed_default_function_attributes
+        end
+
+        def attribute_names_without_changed_default_function_attributes
+          attribute_names.reject do |attr_name|
+            !attribute_changed?(attr_name) && column_for_attribute(attr_name).has_default?
```

Comment:
had to use `#has_default?` instead of `#default_function` because sqlite3 treats `CURRENT_TIMESTAMP` as default value instead of default function.

Reply:
I think SQLite limitation is related to the current [Schema statement implementation](https://github.com/rails/rails/blob/fe4377098b3800c3998f0583549a414b99b72b19/activerecord/lib/active_record/connection_adapters/sqlite3/schema_statements.rb#L143).

Instead of using `has_default?` I suggest to still use `default_function` and just skip the test for Sqlite.
We can extend SQLite in a follow up PR.